src/api/api.c

```diff
@@ -122,10 +122,12 @@ int8_t zp_config_insert(z_config_t *config, unsigned int key, z_string_t value)
     return _zp_config_insert(config, key, value);
 }
 
-z_encoding_t z_encoding_default(void) {
-    return (_z_encoding_t){.prefix = Z_ENCODING_PREFIX_EMPTY, .suffix = _z_bytes_make(0)};
+z_encoding_t z_encoding(z_encoding_prefix_t prefix) {
+    return (_z_encoding_t){.prefix = prefix, .suffix = _z_bytes_make(0)};
 }
```

Comment:
To be consistent with other data structures at the API-level, I would still keep the `z_encoding_t z_encoding_default(void)`.

Reply:
As long as `z_encoding_default` keeps existing (it does, l129, I almost missed it), I'm good with this

Reply:
Indeed, `z_econding_default` has not been remove: just reworked.